web/src/hashchange.js

```diff
@@ -367,6 +371,22 @@ function do_hashchange_overlay(old_hash) {
     if (base === "scheduled") {
         scheduled_messages_overlay_ui.launch();
     }
+    if (base === "user") {
+        const user_id = Number.parseInt(hash_parser.get_current_hash_section(), 10);
+        if (!people.is_known_user_id(user_id)) {
+            $("body").append(render_profile_access_error_model());
+
+            modals.open("profile_access_error_modal", {
+                autoremove: true,
+                on_hide() {
+                    browser_history.exit_overlay();
+                },
+            });
```

Comment:
We should move this into a function that we call from user_profile.

Reply:
I have moved the block of code into a separate function in user_profile and called it accordingly.